Include/monitoring.h

```diff
@@ -0,0 +1,314 @@
+#ifndef Py_MONITORING_H
+#define Py_MONITORING_H
+#ifdef __cplusplus
+extern "C" {
+
+#endif
+
+typedef struct _PyMonitoringState {
+    uint8_t active;
```

Comment:
The size of this is part of the API. 
So even though we aren't going to implement full disabling support in this PR, we probably should add an `uint8_t opaque` field now.

Reply:
Last week I think we were talking about disabling being per CodeLike and not per location.

Reply:
Disabling is per location.
The way it is implemented for code objects is that we maintain the set of events from the previous time the code object was instrumented, and work out what is active from that.
`active = event & (~previous_event | active)` which sets the event if it turned on, or if it stays on and is not disabled.

The problem with doing that here is: where does the event set go? We need to add it to all the APIs, and it must be passed by pointer, so it can be updated.

I think it is much simpler, if less memory efficient, to add the extra byte to the `_PyMonitoringState`.